fhir-path/src/main/java/com/ibm/fhir/path/function/ResolveFunction.java

```diff
@@ -72,6 +78,17 @@ public int getMaxArity() {
     @Override
     public Collection<FHIRPathNode> apply(EvaluationContext evaluationContext, Collection<FHIRPathNode> context, List<Collection<FHIRPathNode>> arguments) {
         Collection<FHIRPathNode> result = new ArrayList<>();
+        FHIRPathNode root = FHIRPathUtil.getSingleton(evaluationContext.getExternalConstant("rootResource"));
+        boolean isBundleContext = root != null && root.type() == FHIRPathType.FHIR_BUNDLE;
+        Map<String, Resource> bundleResources = new HashMap<>();
+        if (isBundleContext) {
+            Bundle bundle = root.asResourceNode().resource().as(Bundle.class);
+            for (Entry e : bundle.getEntry()) {
+                if (e.getResource() != null && e.getFullUrl() != null && e.getFullUrl().hasValue()) {
+                    bundleResources.put(e.getFullUrl().getValue(), e.getResource());
+                }
+            }
+        }
```

Comment:
my assumption is that we set the `rootResource` to the outermost bundle (the one we're evaluating against).
i tried to account for the "bundle-in-a-bundle" case in the algorithm below by continuing to loop until we've reached the top of the tree.

Reply:
We have some helper functions (in FHIRPathUtil) that get the right resource depending on how it might be nested. These are typically stashed in the EvaluationContext.

Reply:
I added the test.  I can think of a couple "interesting edge cases" but they seem so edge to me, that I don't care so much about what our behavior is there.

1. what to do if we have a Bundle thats contained within some other resource that is *not* a bundle (and that outer resource is the root/context).  With the current impl, we will not do local bundle resolution in that case, which seems fine to me.

2. what to do if we have a relative literal reference (e.g. `Patient/1`) in a Bundle.entry where that Bundle is inside another Bundle but the outer Bundle has no fullUrl for this entry.  in this scenario, I think the current algorithm will use the fullUrl from the inner bundle's entry to convert the relative URL into an absolute one.  I don't know if that behavior is desired or not, but again it seems like such an edge case that the impl seems ok to me